torch/linalg/__init__.py

```diff
@@ -2251,3 +2251,42 @@
     >>> torch.dist(Q.mT @ Q, torch.eye(4))
     tensor(6.2158e-07)
 """)
+
+vecdot = _add_docstr(_linalg.linalg_vecdot, r"""
+linalg.vecdot(x, y, *, dim=-1, out=None) -> Tensor
+
+Computes the vector product of two batches of vectors along a dimension.
```

Comment:
There are several "vector products", this one computes the vector _dot_ product

Reply:
Yup. The only thing is that this PR implements \sum x_i y_i, which is not the complex dot product. I opened an issue here https://github.com/data-apis/array-api/issues/356. For now, I'm calling this "the real dot product", which is not standard notation either.

I don't think that there's a standard name for this in the complex case. You'd call it C-linear complex dot product (perhaps) to differentiate it from the R-linear real dot product `Re(x^H y)`...

Leaving it like this for now.